src/dlsproto/client/internal/SuspendableRequest.d

```diff
@@ -636,17 +632,20 @@ struct SuspendableRequest
             // SharedWorking.setDesiredState().)
             ReceivedMessageAction msg_action;
             send_interrupted = false;
-            this.conn.sendReceive(
-                ( in void[] received )
-                {
-                    send_interrupted = true;
-                    msg_action = handle_received_message(received);
-                },
+
+            auto event = this.conn.nextEvent(this.conn.NextEventFlags.Receive,
                 ( conn.Payload payload )
                 {
                     payload.add(control_msg);
                 }
             );
+            assert (event.active != event.active.resumed);
```

Comment:
Actually, it might be better to assert that the event is one of the expected types, rather than asserting what it isn't. In this example, a yield-resumed event would pass the assertion, which is wrong.

Reply:
Yeah, this is what I actually wrote yesterday, than I fixed this up, as I thought that the yield case is guarded internally inside the `nextEvent` method.